src/GiftedChat.tsx

```diff
@@ -232,360 +232,233 @@ export interface GiftedChatState<TMessage extends IMessage = IMessage> {
-    renderAvatarOnTop: false,
-    renderBubble: null,
-    renderSystemMessage: null,
-    onLongPress: null,
-    renderUserName: null,
-    renderMessage: null,
-    renderMessageText: null,
-    renderMessageImage: null,
-    renderMessageVideo: null,
-    renderMessageAudio: null,
-    imageProps: {},
-    videoProps: {},
-    audioProps: {},
-    lightboxProps: {},
-    textInputProps: {},
-    listViewProps: {},
-    renderCustomView: null,
-    isCustomViewBottom: false,
-    renderDay: null,
-    renderTime: null,
-    renderFooter: null,
-    renderChatEmpty: null,
-    renderChatFooter: null,
-    renderInputToolbar: null,
-    renderComposer: null,
-    renderActions: null,
-    renderSend: null,
-    renderAccessory: null,
-    isKeyboardInternallyHandled: true,
-    onPressActionButton: null,
-    bottomOffset: null,
-    minInputToolbarHeight: 44,
-    keyboardShouldPersistTaps: Platform.select({
+function GiftedChat(props: GiftedChatProps) {
+  const {
+    messages = [],
+    messagesContainerStyle = undefined,
+    text = undefined,
+    initialText = '',
+    isTyping,
+    placeholder = DEFAULT_PLACEHOLDER,
+    disableComposer = false,
+    messageIdGenerator = () => uuid.v4(),
+    user = {},
+    onSend = () => {},
+    locale = null,
+    timeFormat = TIME_FORMAT,
+    dateFormat = DATE_FORMAT,
+    loadEarlier = false,
+    onLoadEarlier = () => {},
+    isLoadingEarlier = false,
+    renderLoading = null,
+    renderLoadEarlier = null,
+    renderAvatar = undefined,
+    showUserAvatar = false,
+    actionSheet = null,
+    onPressAvatar = null,
+    onLongPressAvatar = null,
+    renderUsernameOnMessage = false,
+    renderAvatarOnTop = false,
+    renderBubble = null,
+    renderSystemMessage = null,
+    onLongPress = null,
+    renderUserName = null,
+    renderMessage = null,
+    renderMessageText = null,
+    renderMessageImage = null,
+    renderMessageVideo = null,
+    renderMessageAudio = null,
+    imageProps = {},
+    videoProps = {},
+    audioProps = {},
+    lightboxProps = {},
+    textInputProps = {},
+    listViewProps = {},
+    renderCustomView = null,
+    isCustomViewBottom = false,
+    renderDay = null,
+    renderTime = null,
+    renderFooter = null,
+    renderChatEmpty = null,
+    renderChatFooter = null,
+    renderInputToolbar = null,
+    renderComposer = null,
+    renderActions = null,
+    renderSend = null,
+    renderAccessory = null,
+    isKeyboardInternallyHandled = true,
+    onPressActionButton = null,
+    bottomOffset = null,
+    minInputToolbarHeight = 44,
+    keyboardShouldPersistTaps = Platform.select({
       ios: 'never',
       android: 'always',
       default: 'never',
     }),
-    onInputTextChanged: null,
-    maxInputLength: null,
-    forceGetKeyboardHeight: false,
-    inverted: true,
-    extraData: null,
-    minComposerHeight: MIN_COMPOSER_HEIGHT,
-    maxComposerHeight: MAX_COMPOSER_HEIGHT,
-  }
-
-  static propTypes = {
```

Comment:
I removed all propTypes in favor of relying on Typescripts type checking instead.